tests/cuda_custom_call/foo.cu.cc

```diff
@@ -0,0 +1,130 @@
+#include "xla/ffi/api/c_api.h"
+#include "xla/ffi/api/api.h"
+#include "xla/ffi/api/ffi.h"
+
+namespace ffi = xla::ffi;
+
+//----------------------------------------------------------------------------//
+//                            Forward pass                                    //
+//----------------------------------------------------------------------------//
+
+
+// c = a * (b+1)
+// This strawman operation works well for demo purposes because:
+// 1. it's simple enough to be quickly understood,
+// 2. it's complex enough to require intermediate outputs in grad computation,
+//    like many operations in practice do, and
+// 3. it does not have a built-in implementation in JAX.
+__global__ void FooFwdKernel(const float *a,
+                             const float *b,
+                             float *c,
+                             float *b_plus_1,  // intermediate output b+1
+                             size_t n) {
+  size_t tid = blockIdx.x * blockDim.x + threadIdx.x;
+  const size_t grid_stride = blockDim.x * gridDim.x;
+  for (size_t i = tid; i < n; i += grid_stride) {
+    b_plus_1[i] = b[i] + 1.0f;
+    c[i] = a[i] * b_plus_1[i];
+  }
+}
+
+// XLA FFI binding wrapper that launches the kernel
+extern "C" XLA_FFI_Error *FooFwd(XLA_FFI_CallFrame *call_frame) {
+  static const auto *kImpl =
+      ffi::Ffi::Bind()
+          .Ctx<ffi::PlatformStream<cudaStream_t>>()  // stream
+          .Arg<ffi::Buffer<ffi::DataType::F32>>()    // a
+          .Arg<ffi::Buffer<ffi::DataType::F32>>()    // b
+          .Ret<ffi::Buffer<ffi::DataType::F32>>()    // c
+          .Ret<ffi::Buffer<ffi::DataType::F32>>()    // b_plus_1
+          .Attr<size_t>("n")
```

Comment:
What was the argument for using `n` instead of getting the size directly from `a.dimensions`? Just to test that attributes are passed properly?